core/domain/exp_domain.py

```diff
@@ -4017,8 +4015,12 @@ def get_all_html_content_strings(self):
         html_list = []
         for state in self.states.itervalues():
             content_html = state.content.html
-            interaction_html = state.interaction.get_all_html_content_strings()
-            html_list = html_list + [content_html] + interaction_html
+            interaction_html_list = (
+                state.interaction.get_all_html_content_strings())
+            html_list = html_list + [content_html] + interaction_html_list
+
+        html_list = filter(None, html_list)
```

Comment:
I don't understand this part. Why is there a possibility of None in the list? Could we instead prevent adding that at the outset, rather than adding it and filtering it out later?

Reply:
There can be cases while adding html from different parts in interaction, that can be null. So, instead of checking separately for each html, we can filter it at end but we don't need the filtering since any empty html string will not effect the final validation output. So, I have removed the filter.